pkg/sync/queue_manager.go

```diff
@@ -160,6 +160,35 @@ func getQueueKey(run *tektonv1.PipelineRun) string {
 	return fmt.Sprintf("%s/%s", run.Namespace, run.Name)
 }
 
+// FilterPipelineRunByState filters the given list of PipelineRun names to only include those
+// that are in a "queued" state and have a pending status. It retrieves the PipelineRun objects
+// from the Tekton API and checks their annotations and status to determine if they should be included.
+//
+// Returns A list of PipelineRun names that are in a "queued" state and have a pending status.
+func FilterPipelineRunByState(ctx context.Context, tekton versioned2.Interface, orderList []string, wantedStatus, wantedState string) []string {
+	orderedList := []string{}
+	for _, prName := range orderList {
+		prKey := strings.Split(prName, "/")
+		pr, err := tekton.TektonV1().PipelineRuns(prKey[0]).Get(ctx, prKey[1], v1.GetOptions{})
+		if err != nil {
+			continue
+		}
+
+		state, exist := pr.GetAnnotations()[keys.State]
+		if !exist {
+			continue
+		}
+
+		if state == wantedState {
+			if wantedStatus != "" && pr.Spec.Status != tektonv1.PipelineRunSpecStatus(wantedStatus) {
+				continue
+			}
+			orderedList = append(orderedList, fmt.Sprintf("%s/%s", pr.GetNamespace(), pr.GetName()))
```

Comment:
I m assuming `prName` will be in this fomrat prNamspace/prName
if that's the case can we use `prName` instead of `fmt.Sprintf("%s/%s", pr.GetNamespace(), pr.GetName())` :thinking:

Reply:
sure, but i have another PR that add a utility function just for that:

https://github.com/openshift-pipelines/pipelines-as-code/blob/a1a9ba12/pkg/sync/queue_manager_interface.go#L24-L30

i'll submit that PR with this fix you suggested using the helper function after this one since it's a followup and rather keep multiple PR separated focus on small fixes each

Reply:
i did it btw as you suggested in the subsequent iteration... my comment wasn't accurate..